src/Engine/Physics/Trigger.cpp

```diff
@@ -30,12 +34,12 @@ namespace Physics {
 
     void Trigger::Process(btCollisionWorld& world) {
         for (auto& observerPair : observers) {
-            world.contactPairTest(trigger, observerPair.first->GetRigidBody(), *this);
+            world.contactPairTest(trigger, observerPair.first, *this);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'OnEnter' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)